CIP-?/README.md

```diff
@@ -0,0 +1,783 @@
+indicated by the sign of the second argument.
+
+---
+```haskell
+popCountByteString :: BuiltinByteString -> BuiltinInteger
+```
+Returns the number of $1$ bits in the argument.
+
+---
+```haskell
+testBitByteString :: BuiltinByteString -> BuiltinInteger -> BuiltinBool
+```
+If the position given by the second argument is not in
+bounds for the first argument, error; otherwise, if the bit given by that
+position is $1$, return `True`, and `False` otherwise.
+
+---
+```haskell
+writeBitByteString :: BuiltinByteString -> BuiltinInteger -> BuiltinBool -> BuiltinByteString
+```
+If the position given by the second argument is not in bound for the first 
+argument, error; otherwise, set the bit given by that position to $1$ if the 
+third argument is `True`, and $0$ otherwise.
+
+---
+```haskell
+findFirstSetByteString :: BuiltinByteString -> BuiltinInteger
+```
+Return the lowest index such that `testBitByteString` with the first argument 
+and that index would be `True`. If no such index exists, return $-1$ instead.
+
+## Semantics
+
+### Preliminaries
+
+We define $\mathbb{N}^{+} = \\{ x \in \mathbb{N} \mid x \neq 0 \\}$. We assume
+that `BuiltinInteger` is a faithful representation of $\mathbb{Z}$. A
+*bit sequence* $s = s_n s_{n-1} \ldots s_0$ is a sequence such that for
+all $i \in \\{ 0,1,\ldots,n \\}$, $s_i \in \\{ 0, 1 \\}$. A bit sequence 
+$s = s_n s_{n-1} \ldots s_0$ is a *byte sequence* if $n = 8k - 1$ for some 
+$k \in \mathbb{N}$. We denote the *empty bit sequence* (and, indeed, empty byte 
+sequence as well) by $\emptyset$.
+
+We assume that `BuiltinByteString`s represent byte sequences, with the indexes
+of the represented byte sequence being treated in little-endian,
+least-significant-bit-first encoding. For example, consider the byte sequence 
+$s = 110110011100000$; the `BuiltinByteString` literal corresponding to this would
+be `"\217\224"`.
+
+Let $i \in \mathbb{N}^{+}$. 
+We define the sequence $\texttt{binary}(i) = (d_0, m_0), (d_1, m_1), \ldots$ as 
+
+- $m_0 = i \mod 2$, 
+  $d_0 = \frac{i}{2}$ if $i$ is even, 
+  and $\frac{i - 1}{2}$ otherwise.
+- $m_j = d_{j - 1} \mod 2$, 
+  $d_j = \frac{d_{j-1}}{2}$ if $d_j$ is even,
+  and $\frac{d_{j-1} - 1}{2}$ if it is odd.
+
+Some examples follow.
+
+- $\texttt{binary}(4) = (2, 0), (1, 0), (0, 1), (0, 0), (0, 0), \ldots$
+- $\texttt{binary}(17) = (8, 1), (4, 0), (2, 0), (1, 0), (0, 1), (0, 0), (0, 0), \ldots$
+- $\texttt{binary}(553) = (276, 1), (138, 0), (69, 0), (34, 1), (17, 0), (8, 1), (4, 0), (2, 0), (1, 0), (0, 1), (0, 0), (0, 0), \ldots$
+
+### Representation of `BuiltinInteger` as `BuiltinByteString` and conversions
+
+We describe the translation of `BuiltinInteger` into `BuiltinByteString` which 
+is implemented as the `integerToByteString` primitive. Informally, we represent
+`BuiltinInteger`s with the least significant bit at bit position $0$, using a 
+twos-complement representation. More precisely, let $i \in \mathbb{N}^{+}$. We 
```

Comment:
"This corresponds to a little-endian, least-significant-bit first encoding."

Reply:
Amended.